src/main/duke/command/DeadlineCommand.java

```diff
@@ -0,0 +1,39 @@
+package duke.command;
+
+import duke.data.TaskList;
+import duke.data.task.Deadline;
+import duke.storage.Storage;
+import duke.ui.Ui;
+
+/**
+ * This class abstracts the deadline command that the user wants to execute.
+ */
+public class DeadlineCommand extends Command {
+    public static final String COMMAND_WORD = "deadline";
+    private final Deadline newTask;
```

Comment:
Variable newTask implies that there is an old task. Would it be better to name it as task or deadlineTask?